chromium_src/content/browser/speech/network_speech_recognition_engine_impl.cc

```diff
@@ -0,0 +1,41 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include <string>
+
+#include "base/feature_list.h"
+#include "base/metrics/field_trial_params.h"
+#include "url/gurl.h"
+
+namespace {
+
+BASE_FEATURE(kSttFeature, "speech_to_text", base::FEATURE_DISABLED_BY_DEFAULT);
+const base::FeatureParam<std::string> kSttUrl{&kSttFeature, "web_service_url",
+                                              ""};
+
+std::string GetWebServiceBaseUrl(const char* web_service_base_url) {
+  if (base::FeatureList::IsEnabled(kSttFeature)) {
+    return kSttUrl.Get();
+  }
+  // Fallback to Google's url.
+  return web_service_base_url;
+}
+
+}  // namespace
+
```

Comment:
we would also need to guard this with the brave services key

Reply:
also we would need an entry in https://github.com/brave/brave-core/blob/master/chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc but probably after sec review